ladybug/datatype/base.py

```diff
@@ -212,6 +265,74 @@ def _clean(self, unit):
                     ' ', '').replace(
                         '%', 'pct')
 
+    @property
+    def name(self):
+        """The data type name."""
+        if self._name is None:
+            return re.sub(r"(?<=\w)([A-Z])", r" \1", self.__class__.__name__)
+        else:
+            return self._name
+
+    @property
+    def units(self):
+        """A list of all acceptabledata type units."""
+        return self._units
+
+    @property
+    def si_units(self):
+        """A list of acceptable si_units."""
+        return self._si_units
+
+    @property
+    def ip_units(self):
+        """A list of acceptable ip_units."""
+        return self._ip_units
+
+    @property
+    def min(self):
+        """The minimum value of the data type."""
+        return self._min
+
+    @property
+    def max(self):
+        """The maximum possible value of the data type."""
+        return self._max
+
+    @property
+    def abbreviation(self):
+        """The abbreviation of the data type."""
+        return self._abbreviation
+
+    @property
+    def unit_descr(self):
+        """A description of the data type."""
+        return self._unit_descr
+
+    @property
+    def point_in_time(self):
```

Comment:
since cumulative and point_in_time will never happen together can't we just use a single value for both to avoid mistakes.

```python
    @property
    def point_in_time(self):
          return self._point_in_time

    @property
    def cumulative(self):
        """Whether the data type is cumulative."""
        return not self.point_in_time
```

Reply:
I sincerely wish that it could be simplified and it is clear that you wish it too since you have brought this up in every time that you have made comments.

The issue arises from the illuminance in the EPW file, which cannot be cumulative in any meaningful way.  However, the illuminance values in the EPW also do not represent a point in time but rather an average over the previous hour.  So while point in time and cumulative can never both be true, they can both be false and that is why they need to be separate parameters.

We can always go back to having middle_hour as the name of the property that deals with this odd illuminance situation instead of point_int_time if you think that having it called point_in_time name makes it more confusing.  Or we could call it point_in_time_epw if we want to make it clear that this is really only there to deal with the oddities of the epw format.

Reply:
Interesting! Then should we by default have them as a single value and only for the case of illuminance overwrite that?

In any case this is much better than middle_hour.